src/net/server/handlers/query.rs

```diff
@@ -64,13 +64,22 @@ impl<T> Clone for ReservedPermit<T> {
 
 /// Middleware that first reserves a permit on the channel to send messages to the messaging layer.
 /// Once reserved, adds the permit to the extension for retrieval from the handler.
-pub async fn obtain_permit_mw<T: Send + 'static, B>(
-    sender: mpsc::Sender<T>,
-    mut req: Request<B>,
+pub async fn obtain_permit_mw<B: Send>(
+    message_send_map: MessageSendMap,
+    req: Request<B>,
     next: Next<B>,
 ) -> Result<Response, MpcHelperServerError> {
+    let mut req_parts = RequestParts::new(req);
+    let Path(query_id, _) = Path::from_request(&mut req_parts).await?;
```

Comment:
it doesn't seem right to parse the entire path, drop one argument on the floor in middleware and then parse it again in handler only to pick up the dropped argument. 

I guess we need to live with it as axum does not offer anything more convenient that this? Would it work if we inject path as extension here?

Reply:
yeah, i could inject path as an extension. then at least it's only cloned a couple times...